test/cookbooks/acme_test/recipes/acme_dns_client.rb

```diff
@@ -0,0 +1,111 @@
+chef_gem 'acme-client' do
+  compile_time true
+end
+
+#
+# Configure Bind
+#
+
+bind_service 'default' do
+  action [:create, :start]
+end
+
+cookbook_file '/etc/named.conf'
+
+directory '/var/named/master' do
+  recursive true
+end
+
+template '/var/named/master/db.example.org' do
+  source 'db.example.org.erb'
+  notifies :restart, 'bind_service[default]', :immediately
+end
+
+#
+# Setup PostgreSQL server
+#
+
+include_recipe 'osl-postgresql::server'
+
+postgresql_database 'testdb'
+
+postgresql_user 'testuser' do
+  database 'testdb'
+  password 'testpass'
+  action :create
+end
+
+file '/var/lib/pgsql/12/data/pg_hba.conf' do
+  content <<-EOF
+local   all             all                                     trust
+host    all             all             0.0.0.0/0               trust
+  EOF
+  notifies :reload, 'service[postgresql]', :immediately
+end
+
+#
+# Configure acme-dns
+#
+
+execute 'add acme-dns ip' do
```

Comment:
I recommend you use the new `osl_fakenic` resource for this from the base cookbook.